aries/aries_vcx_core/src/wallet/askar/askar_did_wallet.rs

```diff
@@ -0,0 +1,188 @@
+use aries_askar::{
+    crypto::alg::Chacha20Types,
+    kms::{KeyAlg, LocalKey},
+};
+use async_trait::async_trait;
+use public_key::Key;
+
+use super::{
+    askar_utils::local_key_to_bs58_public_key, packing::Packing, rng_method::RngMethod,
+    sig_type::SigType, AskarWallet,
+};
+use crate::{
+    errors::error::{AriesVcxCoreError, AriesVcxCoreErrorKind, VcxCoreResult},
+    wallet::{
+        base_wallet::{did_data::DidData, DidWallet},
+        constants::{DID_CATEGORY, TMP_DID_CATEGORY},
+        structs_io::UnpackMessageOutput,
+        utils::{did_from_key, key_from_base58, seed_from_opt},
+    },
+};
+
+#[async_trait]
+impl DidWallet for AskarWallet {
+    async fn create_and_store_my_did(
+        &self,
+        seed: Option<&str>,
+        _did_method_name: Option<&str>,
+    ) -> VcxCoreResult<DidData> {
+        let mut tx = self.backend.transaction(self.profile.clone()).await?;
+
+        let (did, local_key) = self
+            .insert_key(
+                &mut tx,
+                KeyAlg::Ed25519,
+                seed_from_opt(seed).as_bytes(),
+                RngMethod::RandomDet,
+            )
+            .await?;
+
+        let verkey = local_key_to_bs58_public_key(&local_key)?;
+
+        self.insert_did(&mut tx, &did, DID_CATEGORY, &verkey, None)
+            .await?;
+
+        tx.commit().await?;
+
+        Ok(DidData::new(&did, key_from_base58(&verkey)?))
+    }
+
+    async fn key_for_did(&self, did: &str) -> VcxCoreResult<Key> {
+        let mut session = self.backend.session(self.profile.clone()).await?;
+        let data = self.find_current_did(&mut session, did).await?;
+
+        if let Some(did_data) = data {
+            return Ok(did_data.verkey().to_owned());
+        }
+
+        Err(AriesVcxCoreError::from_msg(
+            AriesVcxCoreErrorKind::WalletRecordNotFound,
+            format!("did not found in wallet: {}", did),
+        ))
+    }
+
+    async fn replace_did_key_start(&self, did: &str, seed: Option<&str>) -> VcxCoreResult<Key> {
+        let mut tx = self.backend.transaction(self.profile.clone()).await?;
+
+        let data = self.find_current_did(&mut tx, did).await?;
```

Comment:
Not using variables only once does not necessarily lead to less lines of code. Consider the following 2 variants of the same method:

```rs
async fn delete_record(&self, category: &str, name: &str) -> VcxCoreResult<()> {
        let mut session = self.backend.session(self.profile.clone()).await?;
        Ok(session.remove(category, name).await?)
}

async fn delete_record(&self, category: &str, name: &str) -> VcxCoreResult<()> {
        Ok(self
            .backend
            .session(self.profile.clone())
            .await?
            .remove(category, name)
            .await?)
}
```

This also leads to passing longer sequences of characters as arguments to method calls, so it requires arguments to be visually parsed first in order to understand what is actually passed into the method, which leads to reading code "from inside out" rather than top to bottom. If that is something you are used to, then yes, it will improve readability.

Reply:
Yeah, I fully admit this is a matter of personal preference.